src/main/java/seedu/address/model/module/Module.java

```diff
@@ -163,6 +193,33 @@ public Module addZoomLink(ZoomLink zoomLink) {
         return new Module(this.getName(), zoomLink, this.gradeTracker, this.tags);
     }
 
+    /**
+     * Adds the zoom link for this module.
+     *
+     * @param key name to indicate the zoom link
+     * @param link zoom link
+     * @return module containing the updated zoom links
+     */
+    public Module addZoomLink(String key, ZoomLink link) {
```

Comment:
Should we create a new instance of the updated Module each time a zoom link is added, or should we update the HashMap accordingly? What could be the advantages of the former implementation?

Reply:
Yes I was also thinking about this. Does updating the content violates immutability though? 

If it does then the current implementation's main advantage would only be that it is consistent with the other method 😅

Reply:
I think we can use the current implementation.